File: src/main/java/com/clearcont/clearcontapp/model/EmpresaGroup.java
package com.clearcont.clearcontapp.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.List;

@Getter
@Setter
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class EmpresaGroup {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @OneToMany(fetch = FetchType.EAGER)
    public List<Empresa> empresas;
}

File: src/main/kotlin/br/com/clearcont/clearcontwebapp/models/EmpresaGroup.kt
package br.com.clearcont.clearcontwebapp.models

import jakarta.persistence.*

@Entity
class EmpresaGroup(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    @OneToMany(fetch = FetchType.EAGER)
    var empresas: MutableList<Empresa> = mutableListOf()
) {
    fun addEmpresa(empresa: Empresa) {
        empresas.add(empresa)
    }

    fun removeEmpresa(empresa: Empresa) {
        empresas.remove(empresa)
    }
}
